### DISCUSSION TITLE
Notice before introducing breaking changes in the future

### DISCUSSION THREAD

**Discussion by luggash** [bluecaret/carettab#349] [Help and Support] [answered]
Hi, I've been using carettab for years now and was very happy with it in the sense that I did not have to think about it after finding a nice design/setup for me because it just worked.

While I understand the technical implications of a major update change in framework (being a front-end dev myself), atleast a small note that there will be breaking changes would have been nice. 
The recent update cleared all my settings, and the settings import does not support backups from previous versions. I am now forced to sleuth through the old settings.json and manually re-creating my new tab page. 

This has happened at least once in the past too, please consider this for any future breaking changes.

I hope this is not the totally wrong place to post this, (re)move if necessary.

Edit: Also, having the color picker locked on HSL format and not being able to textually put in values is a bit of a hassle in my honest opinion.

**Comment by bluecaret** [comment, marked as answer]
I apologize for the trouble. The reset of settings was required to allow more customization and freedom of placement for widgets. I hope you continue giving CaretTab a try. I did put a message in the settings panel a few months ago giving a heads up about this but I realize now that was not enough and apologize. Thank you.
